### ISSUE TITLE
FileHandler Garbage Collector fails to delete expired session files. 

### ISSUE THREAD

**Issue by timothymarois** [codeigniter4/CodeIgniter4#942]
From testing on Amazon Linux EC2 with php 7.0 and php 7.1 the session files are not findable with the current file name detection. (using the default session save path)

I would request this be looked more into, I modified the script to at least get it working on my setup for the time being (example below)

On a higher note, there is something going on with PHP 7+ session gc. From many tests, it is failing to actually clear out the expired session files, I am not entirely sure if this is strictly a CI issue. But it has affected most of my CI projects on different servers. 

**This did not work on the server:** 
```php

$pattern = sprintf('/^%s[0-9a-f]{%d}$/', preg_quote($this->cookieName, '/'), ($this->matchIP === true ? 72 : 40));

while (($file = readdir($directory)) !== false)
{
	// If the filename doesn't match this pattern, it's either not a session file or is not ours
	if ( ! preg_match($pattern, $file) || ! is_file($this->savePath . '/' . $file) || ($mtime = filemtime($this->savePath . '/' . $file)) === false || $mtime > $ts)
	{
		continue;
	}

	unlink($this->savePath . '/' . $file);
}

```

**I ran this instead:** 

```php

while (($file = readdir($directory)) !== false)
{
            $mtime = filemtime($savePath . '/' . $file);

            if (preg_match('|^ci_session|', $file) && $mtime < $ts)
            {
                unlink($savePath . '/' . $file);
            }

}

```

I did not look into which part of the if statement was causing this issue, but I am adding this here as it needs to be investigated further.

**Comment by lonnieezell** [comment]
@timothymarois Can you check your php.ini file on your Amazon server and tell me what the values are for: 

```
session.gc_maxlifetime = 3600
session.gc_probability = 1
session.gc_divisor = 1000
```

**Comment by jim-parry** [comment]
This has come up a number of times on the forum, and appears to be a PHP garbage collector problem, no?

**Comment by lonnieezell** [comment]
Closed via #1684
